test/test_det_noise_sim.cpp

```diff
@@ -0,0 +1,188 @@
+#include "DeterministicNoiseSimulator.hpp"
+#include "nlohmann/json.hpp"
+
+#include "gtest/gtest.h"
+#include <memory>
+
+using namespace dd::literals;
+
+std::unique_ptr<qc::QuantumComputation> getAdder4Circuit() {
+    auto quantumComputation = std::make_unique<qc::QuantumComputation>(4);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 0, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 1, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 3, qc::H);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 2_pc, 3, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 0, qc::T);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 1, qc::T);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 2, qc::T);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 3, qc::Tdag);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 0_pc, 1, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 2_pc, 3, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 3_pc, 0, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 1_pc, 2, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 0_pc, 1, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 2_pc, 3, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 0, qc::Tdag);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 1, qc::Tdag);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 2, qc::Tdag);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 3, qc::T);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 0_pc, 1, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 2_pc, 3, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 3, qc::S);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 3_pc, 0, qc::X);
+    quantumComputation->emplace_back<qc::StandardOperation>(4, 3, qc::H);
```

Comment:
FYI: The QuantumComputation now has helper methods such as `qc->x(qubit)` to add operations in a more readable way. We should use this when writing new tests :)